zrml/orderbook/src/benchmarks.rs

```diff
@@ -50,71 +48,69 @@ fn generate_funded_account<T: Config>(seed: Option<u32>) -> Result<T::AccountId,
     let market = market_mock::<T>();
+    let maker_asset = market.base_asset;
+    let acc = generate_funded_account::<T>(seed, maker_asset)?;
+    let maker_amount: BalanceOf<T> = 100u32.into();
+
+    let taker_amount: BalanceOf<T> = BASE.saturated_into();
     T::MarketCommons::push_market(market.clone()).unwrap();
-    Ok((acc, outcome_asset, outcome_asset_amount, base_asset_amount, market_id))
+    let market_id: MarketIdOf<T> = 0u32.into();
+    Ok((market_id, acc, maker_asset, maker_amount, taker_amount))
 }
 
 // Creates an order of type `order_type`. `seed` specifies the account seed,
 // None will return a whitelisted account
 // Returns `account`, `asset`, `order_id`
-fn place_order<T: Config>(
-    order_type: OrderSide,
+fn place_default_order<T: Config>(
     seed: Option<u32>,
+    taker_asset: AssetOf<T>,
 ) -> Result<(T::AccountId, MarketIdOf<T>, OrderId), &'static str> {
-    let (acc, outcome_asset, outcome_asset_amount, base_asset_amount, market_id) =
+    let (market_id, acc, maker_asset, maker_amount, taker_amount) =
         order_common_parameters::<T>(seed)?;
 
     let order_id = <NextOrderId<T>>::get();
-    let _ = Call::<T>::place_order {
-        market_id,
-        outcome_asset,
-        side: order_type.clone(),
-        outcome_asset_amount,
-        base_asset_amount,
-    }
-    .dispatch_bypass_filter(RawOrigin::Signed(acc.clone()).into())?;
+    let _ =
+        Call::<T>::place_order { market_id, maker_asset, maker_amount, taker_asset, taker_amount }
+            .dispatch_bypass_filter(RawOrigin::Signed(acc.clone()).into())?;
 
     Ok((acc, market_id, order_id))
 }
 
 benchmarks! {
-    remove_order_ask {
-        let (caller, _, order_id) = place_order::<T>(OrderSide::Ask, None)?;
+    remove_order {
+        let taker_asset = Asset::CategoricalOutcome::<MarketIdOf<T>>(0u32.into(), 0);
+        let (caller, _, order_id) = place_default_order::<T>(None, taker_asset)?;
     }: remove_order(RawOrigin::Signed(caller), order_id)
 
-    remove_order_bid {
-        let (caller, _, order_id) = place_order::<T>(OrderSide::Bid, None)?;
-    }: remove_order(RawOrigin::Signed(caller), order_id)
-
-    fill_order_ask {
-        let caller = generate_funded_account::<T>(None)?;
-        let (_, _, order_id) = place_order::<T>(OrderSide::Ask, Some(0))?;
-    }: fill_order(RawOrigin::Signed(caller), order_id, None)
-
-    fill_order_bid {
-        let caller = generate_funded_account::<T>(None)?;
-        let (_, _, order_id) = place_order::<T>(OrderSide::Bid, Some(0))?;
+    fill_order {
+        let taker_asset = Asset::CategoricalOutcome::<MarketIdOf<T>>(0u32.into(), 0);
+        let caller = generate_funded_account::<T>(None, taker_asset)?;
+        let (_, market_id, order_id) = place_default_order::<T>(Some(0), taker_asset)?;
+        let maker_asset = T::MarketCommons::market(&market_id).unwrap().base_asset;
+        let caller = generate_funded_account::<T>(None, maker_asset)?;
```

Comment:
It's not a major problem. But even now it's really weird design, because you're now handling two markets IDs and solving the problem by just throwing one of them away.

Reply:
I admit, I don't really give a high attention to the design of benchmarks as soon as they are quick to write. What do you propose us to do now. Can you suggest something or should we just ignore this issue?

Reply:
No just leave it